src/ussd_popi_rapidpro.js

```diff
@@ -927,37 +927,194 @@ go.app = function() {
             });
         });
 
-        self.add("state_no_optout", function(name) {
+        self.add("state_anonymous_data", function(name) {
             return new MenuState(name, {
-                question: $(
-                    "Thanks! MomConnect will continue to send helpful messages and process " +
-                    "your personal info. What would you like to do?"
+                question: $("If you make your data anonymous, we'll delete your phone number, " +
+                            "and we won't be able to send you messages." ,
+                            " " ,
+                            "Do you want to continue?"),
+                error: $(
+                    "Sorry we don't recognise that reply. Please enter the number next to your " +
+                    "answer."
                 ),
+                choices: [
+                    new Choice("state_anonymous_data_optout", $("Yes")),
+                    new Choice("state_optout_menu", $("No"))
+                ]
+            });
+        });
+
+        self.add("state_no_optout", function(name) {
+            return new EndState(name, {
+                next: "state_start",
+                text: $(
+                    "Thanks! MomConnect will continue to send you helpful messages. " +
+                    "Have a lovely day!"
+                )
+            });
+        });
+
+        self.add("state_anonymous_data_optout", function(name, opts){
+            var msisdn = utils.normalize_msisdn(self.im.user.addr, "ZA");
+            var answers = self.im.user.answers;
+            var opt_out = _.toUpper(answers.state_stop_research) === "YES";
+            return self.rapidpro
+                .start_flow(
+                    self.im.config.optout_flow_id, null, "whatsapp:" + _.trim(msisdn, "+"), {
+                        opt_out: opt_out ? "TRUE" : "FALSE",
+                        mqr_consent: "Denied",
+                        research_consent: "FALSE"
+                    }
+                )
+                .then(function() {
+                    return self.states.create("state_opt_out_reason");
+                }).catch(function(e) {
+                    console.log(e.message);
+                    // Go to error state after 3 failed HTTP requests
+                    opts.http_error_count = _.get(opts, "http_error_count", 0) + 1;
+                    if(opts.http_error_count === 3) {
+                        self.im.log.error(e.message);
+                        return self.states.create("__error__", {return_state: name});
+                    }
+                    return self.states.create(name, opts);
+                });
+        });
+
+        self.add("state_stop_research_optout", function(name, opts){
+            var msisdn = utils.normalize_msisdn(self.im.user.addr, "ZA");
+            return self.rapidpro
+                .start_flow(
+                    self.im.config.clear_pii_flow, null, "whatsapp:" + _.trim(msisdn, "+")
```

Comment:
Instead of `clear_pii_flow` lets use something like `research_optout_flow`, we can create a new flow called `External Opt Out - Research` for this in rapidpro where we just change mqr_consent to denied and research_consent to FALSE

Reply:
Looks like this one is also not resolved